docs/upgrade_guide.md

```diff
@@ -0,0 +1,202 @@
+
+```json
+[Unit]
+Description=Cyber Node
+After=network-online.target
+
+[Service]
+User=ubuntu
+WorkingDirectory=/home/ubuntu/.cyberd/
+ExecStart=/home/ubuntu/.cyberd/cosmosd start --compute-rank-on-gpu=true
+Environment=DAEMON_HOME=/home/ubuntu/.cyberd
+Environment=DAEMON_NAME=cyberd
+Environment=GAIA_HOME=/home/ubuntu/.cyberd
+Environment=DAEMON_RESTART_AFTER_UPGRADE=on
+Restart=always
+RestartSec=3m
+LimitNOFILE=4096
+
+[Install]
+WantedBy=multi-user.target
+```
+
+The most important part is
+
+```bash
+Environment=DAEMON_RESTART_AFTER_UPGRADE=on
+```
+
+Because it was missing in out earlier versions of run validator guide, and higly important for node start after upgrade.
+
+If service file was modified, systemd must be reloaded to apply changes:
+
+```bash
+systemctl daemon-reload
+```
+
+### Prepraing new cyberd binary
+
+Upgrade will be released with tag v0.1.6.3, so technically all that required to get it is pull changes inside cloned go-cyber repo:
+
+```bash
+cd /<path_to_go-cyber>/go-cyber/
+git reset --hard 
+git checkout master
+git pull
+```
+
+and then build all in the same way as during setup, cuda code first, `cyberd` binary next:
+
+```bash
+cd /<path_to_go-cyber>/go-cyber/x/rank/cuda/
+make
+cd /<path_to_go-cyber>/go-cyber/
+make build
+```
+
+After all above done, new `cyberd` and `cyberdcli` binaries must appear inside ~/go-cyber/build/.
+
+Next we need to create folders tree with upgrade name and place binaries to appropriate location:
+
+```bash
+mkdir $DAEMON_HOME/upgrade_manager/upgrades/
+mkdir $DAEMON_HOME/upgrade_manager/upgrades/darwin/
+mkdir $DAEMON_HOME/upgrade_manager/upgrades/darwin/bin/
+cp build/cyberd $DAEMON_HOME/upgrade_manager/upgrades/darwin/bin/
+cp build/cyberdcli /usr/local/bin/
+cp build/cyberd /usr/local/bin/
+chmod +x $DAEMON_HOME/upgrade_manager/upgrades/darwin/bin/cyberd
+```
+
+To check that everything builded correclty run:
+
+```bash
+$DAEMON_HOME/upgrade_manager/upgrades/darwin/bin/cyberd version
+```
+
+Version must be `v0.1.6.3`.
+
+And verify upgrade_manager folder from .cyberd directory looks like this:
+
+```bash
+.
+├── current -> /root/.cyberd/genesis/bin
+├── genesis
+│   └── bin
+│       └── cyberd
+└── upgrades
+    └── darwin
+        └── bin
+            └── cyberd
+```
+
+And here thats it for preparation side. It everythng set correctly upgrade will happen automatically.
+
+## Upgrade itself
+
+When upgrade proposal will be submited and upgrade block will come, cosmosd will do following on all nodes across the chain:
+
+- Stop cyber node
+
+- Change `current` link to` -> .cyberd/upgrade_manager/upgrades/darwin`
+
+- Automatically start node
+
+As soon as there will be more than ~66.66% of voting power at succeded validators chain will continue block production.
+
+## Manual upgrade
+
+If something is goes wrong and cosmosd didn't restart with new binary at given time on upgrade you need do this:
+...
```

Comment:
@mrlp4 need to add actions which need to be performed manually
also add note that upgrades not covered by docker setup